src/buffer_params.rs

```diff
@@ -0,0 +1,261 @@
+use crate::context_handle::PtrWrap;
+use crate::enums::CapStyle;
+use crate::{
+    AsRaw, AsRawMut, ContextHandle, ContextHandling, ContextInteractions, Error, GResult, JoinStyle,
+};
+
+use geos_sys::*;
+use std::sync::Arc;
+
+pub const DEFAULT_MITRE_LIMIT: f64 = 5.0;
+pub const DEFAULT_QUADRANT_SEGMENTS: i32 = 8;
+pub const DEFAULT_SINGLE_SIDED: bool = false;
```

Comment:
That seems like they should all be enums.

Reply:
You would have to explain - I can't see how an enum would fit here. 
The constants are already namespaced by the `buffer_params` module. If those constants became an enum we would then have to create a separate instance of it as the individual cases can't have defaults (a single value) without being a C style enum. Even then a C-style enum is not well typed unless all cases are integers. In usage, such as `buffer_with_style` the cases are not exclusive/disjoint. The C API has separate set param functions (mitre, cap style...) and not a single setParam that can take a union of inputs.

Reply:
Because I assume there are other constants like `*_MITRE_*` and for others too.

Reply:
You want the enum to behave like a "constants" namespace instead of just using the buffer params module as the namespace? Seems like the actual constant definitions would then have to be a hashmap mapping from the constant type mitre | quad segments | etc to the constant value.

Reply:
Ok, so after taking a look around, I'm not sure if we should keep these constants... If we want to handle default values, might be better to implement `Default` on `BufferParams` instead, no?

Reply:
I can change the `BufferParamsBuilder` to use options instead of default values and only call the C Api to set the values if the option is `Some`...and remove those constants. 
`BufferParams` doesn't really need default as it has `new`. Not setting any options will use the defaults defined in the  C++ API.

Reply:
Fine by me!